tensorflow/python/keras/applications/efficientnet_weight_update_util.py

```diff
@@ -0,0 +1,307 @@
+        for v in v_all:
+          if v.name == tf_name:
+            v_val = sess.run(v)
+            with h5py.File(path_h5, 'a') as f:
+              v_prev = f[keras_name][()]
+              try:
+                f[keras_name].write_direct(v_val)
+              except:
+                raise ValueError(f'weight in {tf_name} does not ift into {keras_name}')
+            print(f'writing from: {tf_name}\n  to: {keras_name}')
+            print(f'  average change: {abs(v_prev - v_val).mean()}')
+            v_all.remove(v)
+            break
+
+
+def model_name_keras_to_tf(model_name_keras):
+  """Infer model name in both keras and tf implementations"""
+  model_name_tf = model_name_keras.replace('efficientnet', 'efficientnet-')
+  return model_name_tf
+
+
+def get_h5_names(path_h5):
+  """Get list of variable names from the h5 file """
+  h5_namelst = []
+  def append_to_lst(x):
+    h5_namelst.append(x)
+
+  with h5py.File(path_h5, 'r') as f:
+    for x in f.keys():
+      f[x].visit(append_to_lst)
+
+  # all weights end with ':0'
+  h5_namelst = [x for x in h5_namelst if ':' in x]
+
+  # append group name to the front
+  h5_namelst = ['/'.join([x.split('/')[0], x]) for x in h5_namelst]
+    
+  return h5_namelst
+
+
+def get_tf_names(path_ckpt, use_ema=True):
+  """Get list of tensor names from checkpoint"""
+
+  tf2_listvar = tf.train.list_variables(path_ckpt)
+
+  if use_ema:
+    tf2_listvar = [x for x in tf2_listvar if 'ExponentialMovingAverage' in x[0]]
+  else:
+    tf2_listvar = [x for x in tf2_listvar if 'ExponentialMovingAverage' not in x[0]]
+
+  # remove util variables used for RMSprop
+  tf2_listvar = [x for x in tf2_listvar if 'RMS' not in x[0]]
+
+  tf2_listvar = [x[0] for x in tf2_listvar]
+  return tf2_listvar
+
+
+def get_tf_blocks(tf_weights):
+  """Extract the block names from list of full weight names"""
+  tf_blocks = set([x.split('/')[1] for x in tf_weights if 'block' in x])
+  tf_blocks = sorted(tf_blocks, key=lambda x:int(x.split('_')[1]))
+  return tf_blocks
+
+
+def get_keras_blocks(keras_weights):
+  """Extract the block names from list of full weight names"""
+  return sorted(set([x.split('_')[0] for x in keras_weights if 'block' in x]))
+
+
+def keras_name_to_tf_name_stem_top(keras_name, use_ema=True, model_name_tf='efficientnet-b0'):
+  """ map name in h5 to ckpt that is in stem or top (head)
+  
+  we map name keras_name that points to a weight in h5 file 
+  to a name of weight in ckpt file. 
+  
+  Args:
+    keras_name: str, the name of weight in the h5 file of keras implementation
+    use_ema: Bool, use the ExponentialMovingAverage resuolt in ckpt or not 
+    model_name_tf: str, the name of model in ckpt.
+
+  Returns:
+    String for the name of weight as in ckpt file.
+
+  Raises:
+    KeyError if we cannot parse the keras_name
+  """
+  if use_ema:
+    ema = '/ExponentialMovingAverage'
+  else:
+    ema = ''
+
+  stem_top_dict = {
+      'probs/probs/bias:0':f'{model_name_tf}/head/dense/bias{ema}:0',
```

Comment:
Please add space before and after ":" so that  it is more clear to find the key and value.